internal/providers/system/system.go

```diff
@@ -56,3 +71,31 @@ func fetchConfig(logger *log.Logger, filename string) (types.Config, report.Repo
 	}
 	return util.ParseConfig(logger, rawConfig)
 }
+
+// fetchBaseConfig is a helper function to merge all the base config fragments inside of a particular directory.
+func fetchBaseConfig(logger *log.Logger, dir string) (types.Config, report.Report, error) {
+	var baseConfig types.Config
+	var ret report.Report
```

Comment:
There are multiple return values so `ret` is ambiguous.  Let's put `report` into the variable name.